Lib/test/test_typing.py

```diff
@@ -3809,6 +3809,26 @@ class Y(C[int]):
         self.assertEqual(Y.__qualname__,
                          'GenericTests.test_repr_2.<locals>.Y')
 
+    def test_repr_3(self):
+        T = TypeVar('T')
+        P = ParamSpec('P')
+
+        class MyCallable(Generic[P, T]):
+            pass
+
+        self.assertEqual(
+            repr(MyCallable[[], bool]).split('.')[-1],
+            'MyCallable[[], bool]',
+        )
+        self.assertEqual(
+            repr(MyCallable[[int], bool]).split('.')[-1],
+            'MyCallable[[int], bool]',
+        )
+        self.assertEqual(
+            repr(MyCallable[[int, str], bool]).split('.')[-1],
+            "MyCallable[[int, str], bool]",
+        )
```

Comment:
It took me a few seconds to grok what these tests were asserting, and why we needed the call to `.split()`. Maybe something like this instead:

```suggestion        
        object_to_expected_repr = {
            MyCallable[[], bool]:         "MyCallable[[], bool]",
            MyCallable[[int], bool]:      "MyCallable[[int], bool]",
            MyCallable[[int, str], bool]: "MyCallable[[int, str], bool]"
        }

        for obj, expected_repr in object_to_expected_repr.items():
            with self.subTest(obj=obj, expected_repr=expected_repr):
                self.assertEqual(repr(obj), MyCallable.__module__ + expected_repr)
```

Reply:
Yeah, it is way nicer!